client/mainmenu/CCampaignScreen.cpp

```diff
@@ -67,7 +67,8 @@ CCampaignScreen::CCampaignScreen(const JsonNode & config, std::string name)
 	}
 
 	for(const JsonNode & node : config[name]["items"].Vector())
-		campButtons.push_back(std::make_shared<CCampaignButton>(node, config, campaignSet));
+		if(CResourceHandler::get()->existsResource(ResourcePath(node["file"].String(), EResType::CAMPAIGN)))
+			campButtons.push_back(std::make_shared<CCampaignButton>(node, config, campaignSet));
```

Comment:
We have logic that if some of references h3c's are missing then corresponding campaign screen won't be available.
Check references to `vcmi.client.errors.missingCampaigns` string ID.

I think right now this check is pointless / unreachable - player can't open such campaign screen.

Reply:
Yes. Know this. It's already checked before entering the screen:
https://github.com/vcmi/vcmi/blob/719859cc62bbcd54dc3f5541bca697f5df8706b3/client/mainmenu/CMainMenu.cpp#L428

"optional" parameter is exactly to avoid to make check useless.